src/Logger.js

```diff
@@ -0,0 +1,115 @@
+ */
+module.exports = class Logger {
+    /**
+     * @param {string} loglevel  Accepted values: fatal | error | warn | info | debug | trace
+     * @param {object} context   Common props to add to all logged messages.
+     */
+    constructor(loglevel = "info", context = {}) {
+        this._logger = pino({
+            messageKey: "message",
+            base:       null,
+            timestamp:  true,
+            level:      loglevel
+        });
+        this.setContext(context);
+    }
+
+    /**
+     * Sets the pino-compatible log level to be used from this point forward.
+     *
+     * @param {string} loglevel  Accepted values: fatal | error | warn | info | debug | trace
+     */
+    setLogLevel(loglevel) {
+        this._logger.level = loglevel;
+    }
+
+    /**
+     * Sets common props that will be added to all logs from this point forward.
+     * Useful for things like request_id, type/namespace/service, etc.
+     *
+     * @param {object} context   Common props to add to all logged messages.
+     */
+    setContext(context) {
+        this._context = context;
+    }
+
+    /**
+     * Adds/overwrites common props that will be added to all logs from this point forward.
+     * Useful for things like request_id, type/namespace/service, etc.
+     *
+     * @param {object} context   Common props to add to all logged messages.
+     */
+    addContext(context) {
+        this._context = {...this._context, ...context};
+    }
+
+    /**
+     * Creates a child logger from the current one, inherting all of parent logger context.
+     *
+     * @param {object} context   Common props to add to current context.
+     */
+    child(context) {
+        return new Logger(this._logger.level, {...this._context, ...context});
+    }
+
+    info(message, props = {}) {
+        this._log("info", message, props);
+    }
+
+    warn(message, props = {}) {
+        this._log("warn", message, props);
+    }
+
+    error(message, props = {}) {
+        if (props instanceof Error) {
+            var error_props = {
+                error_message: props.message
+            };
+            if (props.hasOwnProperty("code")) {
+                error_props.error_code = props.code;
+            }
+            if (props.hasOwnProperty("status")) {
+                error_props.error_status = props.status;
+            }
+            if (props.hasOwnProperty("stack")) {
+                error_props.error_stack = props.stack;
+            }
+            props = error_props;
+        }
+        this._log("error", message, props);
+    }
+
+    debug(message, props = {}) {
+        this._log("debug", message, props);
+    }
+
+    /**
+     * Helper method to record access logs. We want these with no message, context: "access" and
+     * not casting all props as strings, since we use a few integers there and the format is standard.
+     */
+    access(props = {}) {
+        this._log("info", undefined, props, "access", false);
```

Comment:
I believe this won't be logged if the log level is higher than `info`, because you share the same pino instance between access and app logs. Since access logs needs to be logged no matter which level is set, you should check if there's a way to make an override, or you'll probably have to use two different pino instances.

Reply:
I looked in the pino docs and the only way to "override" the log level would be to do something like
```
var level = this._logger.level;
this.setLogLevel("info");
this.access(...);
this.setLogLevel(level);
```
which seems kind of hacky to me, so I opted for having a separate logger that gets created only if we need to write access logs. Let me know if you like this approach or prefer the `setLogLevel` one